### PR TITLE
Add default rollup config and rollup config to core package

### PR DESCRIPTION
**Pull request by bertya**
Basic Rollup config file is being added to root. And the core package imports it. For other packages, they could import it and modify it.

**Tested with a simple react app created via creat-react-app, components work fine. Only issue is that pcln-icons package seems to be a peer dependency.**

for size comparison, in that simple react app, it imports like this in app.js
`import { Box, Banner, ThemeProvider, ToggleBadge } from 'pcln-design-system'`

With rollup esm export 
```
File sizes after gzip:

  72.59 KB  build/static/js/1.38ac86de.chunk.js
  7.92 KB   build/static/js/main.08411e1d.chunk.js
  763 B     build/static/js/runtime~main.229c360f.js
  510 B     build/static/css/main.0fe55b7a.chunk.css
```

With design system v1
```
File sizes after gzip:

  116.06 KB (+43.47 KB)  build/static/js/1.adb2dba3.chunk.js
  808 B (-7.13 KB)       build/static/js/main.33990733.chunk.js
  763 B                  build/static/js/runtime~main.229c360f.js
  511 B (+1 B)           build/static/css/main.7eb40025.chunk.css
```

### CODE REVIEW

**Comment by jxnblk** [comment]
Awesome! The bundle size reduction looks promising! I assume this is a replacement for #421?

File: rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const resolve = require('rollup-plugin-node-resolve')
+const babel = require('rollup-plugin-babel')
+const commonjs = require('rollup-plugin-commonjs')
+const json = require('rollup-plugin-json')
+const fileSize = require('rollup-plugin-filesize')
+
+module.exports = {
```
Comment (jxnblk) [inline rollup.config.js]: This looks a lot more *by-the-book* 👍

File: packages/core/package.json [hunk 2]
```diff
@@ -3,8 +3,9 @@
   "version": "2.0.0-beta.7",
   "description": "Priceline Design System",
   "main": "dist/index.js",
+  "module": "dist/my-package.esm.js",
```
Comment (jxnblk) [inline packages/core/package.json]: Is the filename a typo here?

**Comment by hakimelek** [comment]
Should resolve #401.

File: packages/core/package.json (hunk 2 above)
Comment (bertya) [reply packages/core/package.json]: yeah, updated.

**Comment by bertya** [comment]
> Awesome! The bundle size reduction looks promising! I assume this is a replacement for #421?

Yep

**Pull request merged**